packages/babel-plugin-react-jsx/__tests__/TransformJSXToReactJSX-test.js

```diff
@@ -29,15 +30,307 @@ function transform(input, options) {
           {
             useBuiltIns: true,
             useCreateElement: false,
-            ...options,
+            ...pluginOpts,
           },
         ],
       ],
+      ...babelOpts,
     }).code
   );
 }
 
 describe('transform react to jsx', () => {
+  it('throws error when sourceType is module and autoImport is require', () => {
+    let _error;
+    const code = `var x = <div><span /></div>`;
+    try {
+      transform(code, {
+        autoImport: 'require',
+      });
+    } catch (error) {
+      _error = error;
+    }
+    expect(_error).toEqual(
```

Comment:
Did you test on the error here in this manner because expect(...).toThrow(...) was not sufficient to test the shape of this error?